lib/src/kvs/sled/bench.rs

```diff
@@ -0,0 +1,193 @@
+#[cfg(test)]
+mod tests {
+	use crate::kvs::{Key, Val};
+	use rand::Rng;
+	use std::collections::{BTreeMap, HashSet};
+	use std::time::SystemTime;
+
+	const N: usize = 1000;
+	const M: usize = 1000;
+	const KEY: Key = Key::new();
+	const VAL: Val = Val::new();
+
+	fn get_randomized_map(random: u16) -> BTreeMap<Key, Val> {
+		let mut map = BTreeMap::<Key, Val>::new();
+		if random == 0 {
+			map.insert(KEY, VAL);
+		}
+		map
+	}
+
+	fn get_some_randomized_map(random: u16) -> Option<BTreeMap<Key, Val>> {
+		if random == 1 {
+			None
+		} else {
+			Some(get_randomized_map(random))
+		}
+	}
+
+	fn get_none_randomized_map(random: u16) -> Option<BTreeMap<Key, Val>> {
+		if random == 2 {
+			Some(get_randomized_map(random))
+		} else {
+			None
+		}
+	}
+
+	/// This test justifies the rationale of enclosing the `BTreeMap` in an `Option`.
+	/// Mostly, this test evaluates the performance of `Option::is_none` versus `BTreeMap::is_empty`.
+	/// The test passes if `is_none` is at least 2 times faster than `is_empty`.
+	///
+	/// We also check the cost of encapsulating the new map in an Option vs just using the new map.
+	/// The test passes if the difference is less than 10%.
+	///
+	/// This test checks only the time factor benefit. But there is also a space factor benefit
+	/// as the `BTreeMap` structure contains several properties.
+	///
+	/// Note that the `BTreeMap` is also using a similar approach by encapsulating the root of the
+	/// map in an Option: `root: Option<Root<K, V>>`.
+	///
+	#[tokio::test]
+	async fn test_transaction_btreemap_vs_option_rationale() {
+		let mut rng = rand::thread_rng();
+		let rnd: u16 = rng.gen();
+
+		// Evaluating BTreeMap creation + BTreeMap::is_empty()
+		let time = SystemTime::now();
+		for _ in 0..N {
+			let map = get_randomized_map(rnd);
+			for _ in 0..M {
+				assert!(map.is_empty() || map.contains_key(&KEY));
+			}
+		}
+		let map_time = time.elapsed().unwrap();
+
+		// Evaluating Option creation + Option::is_none()
+		let time = SystemTime::now();
+		for _ in 0..N {
+			let map = get_none_randomized_map(rnd);
+			for _ in 0..M {
+				match &map {
+					None => {
+						assert!(true)
```

Comment:
This will always pass. So why not just

```rust
if let  Some(map) = &map {
    // We consume the map to avoid any optimisation
    assert!(map.is_empty() || map.contains_key(&KEY));
}
```

? Is it a trick to avoid optimisation as well?

Reply:
Definitely useless, it was an artefact of several iterations in the code... switching to a `let` statement!